crates/astria-sequencer/src/app.rs

```diff
@@ -146,21 +146,30 @@ pub(crate) struct App {
 }
 
 impl App {
-    pub(crate) fn new(snapshot: Snapshot) -> Self {
+    pub(crate) async fn new(snapshot: Snapshot) -> anyhow::Result<Self> {
         tracing::debug!("initializing App instance");
 
+        let app_hash: AppHash = snapshot
```

Comment:
will this fail if init_chain has not been called yet?

Reply:
no, it'll just return the empty root hash. confirmed that syncing from genesis with an empty state works